### PR TITLE
Edit group chat refactor

### PR DESCRIPTION
**Pull request by foodstr**

### CODE REVIEW

File: UI/app_update.tsx [hunk 1]
```diff
@@ -243,23 +243,19 @@ export async function* UI_Interaction_Update(args: {
...
         //
-        else if (event.type == "EditMyProfile") {
-            model.myProfile = Object.assign(model.myProfile || {}, event.profile);
-        } else if (event.type == "SaveMyProfile") {
-            InsertNewProfileField(app.model);
+        else if (event.type == "SaveProfile") {
+            if (!event.profile) {
+                console.log("impossible");
+                continue;
+            }
```
Comment (BlowaterNostr) [inline UI/app_update.tsx:252]: Why is this even existing?

File: UI/app_update.tsx [hunk 2]
```diff
@@ -38,14 +37,15 @@ import { EventDetail, EventDetailItem } from "./event-detail.tsx";
 import { CreateGroup, CreateGroupChat, StartCreateGroupChat } from "./create-group.tsx";
 import { prepareNormalNostrEvent } from "../lib/nostr-ts/event.ts";
 import { PublicKey } from "../lib/nostr-ts/key.ts";
-import { NostrAccountContext, NostrEvent, NostrKind } from "../lib/nostr-ts/nostr.ts";
+import { InMemoryAccountContext, NostrAccountContext, NostrEvent, NostrKind } from "../lib/nostr-ts/nostr.ts";
```
Comment (BlowaterNostr) [inline UI/app_update.tsx:40]: Used?

File: UI/edit-profile.tsx [hunk 3]
```diff
@@ -1,323 +1,270 @@
...
-    type: "EditMyProfile";
-    profile: ProfileData;
-};
-
-type Save = {
-    type: "SaveMyProfile";
-    profile: ProfileData;
+export type SaveProfile = {
+    type: "SaveProfile";
+    profile: ProfileData | undefined;
```
Comment (BlowaterNostr) [inline UI/edit-profile.tsx:28]: undefined? Meaning?

File: UI/edit-profile.tsx [hunk 4]
```diff
@@ -1,323 +1,270 @@
...
+            profile: profileGetter.getProfilesByPublicKey(ctx.publicKey)?.profile,
+        });
+    }
+
+    shouldComponentUpdate(_: Readonly<Props>, nextState: Readonly<State>, __: any): boolean {
+        return JSON.stringify(this.state.profile) != JSON.stringify(nextState.profile);
+    }
+
+    newFieldKey = createRef<HTMLInputElement>();
+    newFieldValue = createRef<HTMLTextAreaElement>();
```
Comment (BlowaterNostr) [inline UI/edit-profile.tsx:90]: Why do we need ref here?

Comment (foodstr) [reply UI/edit-profile.tsx:90]: we can get the input value from ref

**Review by BlowaterNostr — APPROVED** [review]

**Pull request merged**